src/org/labkey/targetedms/folderImport/PanoramaQCSettings.java

```diff
@@ -0,0 +1,535 @@
+                else
+                    filter.addCondition(FieldKey.fromParts(col), row.get(col));
+            }
+            logMsg += col + ": '" + row.get(col) + "'" + (count == row.size()-1 ? "" : ", ");
+            ++count;
+        }
+        if (new TableSelector(ti, row.keySet(), filter, null).getRowCount() > 0)
+        {
+            logMsg += "] values already exists. Skipping";
+            ctx.getLogger().info(logMsg);
+        }
+        else
+        {
+            dataWithoutDuplicates.add(row);
+        }
+    }
+
+    public int importSettingsFromFile(FolderImportContext ctx, VirtualFile panoramaQCDir, @Nullable TargetedMSSchema schema, @Nullable TableInfo ti, @Nullable QueryUpdateService qus, @Nullable BatchValidationException errors) throws Exception
+    {
+        return importSettings(ctx, ti, _settingsFileName, panoramaQCDir, errors, qus);
+    }
+
+    protected int importSettings(FolderImportContext ctx, TableInfo tinfo, String settingsFileName, VirtualFile panoramaQCDir, @Nullable BatchValidationException errors, @Nullable QueryUpdateService qus) throws Exception
+    {
+        List<Map<String, Object>> tsvData = getTsvData(panoramaQCDir, settingsFileName);
+        List<Map<String, Object>> dataWithoutDuplicates = new ArrayList<>();
+
+        tsvData.forEach(row -> getDataWithoutDuplicates(ctx, tinfo, row, dataWithoutDuplicates));
+        return insertData(ctx.getUser(), ctx.getContainer(), dataWithoutDuplicates, errors, qus, getSettingsFileName(), getTableName());
+    }
+
+    protected List<Map<String, Object>> getTsvData(VirtualFile panoramaQCDir, String settingsFile) throws IOException
+    {
+        try (BufferedReader reader = Readers.getReader(panoramaQCDir.getInputStream(settingsFile)); DataLoader loader = new TabLoader(reader, true))
+        {
+            return loader.load();
+        }
+    }
+
+    protected int insertData(User user, Container container, List<Map<String, Object>> dataWithoutDuplicates, BatchValidationException errors,
+                             QueryUpdateService qus, String settingsFileName, String tableName) throws Exception
+    {
+        if (dataWithoutDuplicates.size() > 0)
+        {
+            if (qus != null)
+            {
+                List<Map<String,Object>> insertedRows;
+                try
+                {
+                    insertedRows = qus.insertRows(user, container, dataWithoutDuplicates, errors, null, null);
+                }
+                catch (DuplicateKeyException | BatchValidationException | QueryUpdateServiceException | SQLException e)
+                {
+                    throw new Exception("Data from " + settingsFileName + " did not get imported into targetedms." + tableName, e);
+                }
+                return insertedRows.size();
+            }
+            else
+            {
+                throw new NotFoundException("Query update service for targetedms." + tableName + " not found.");
```

Comment:
NotFoundException is intended for HTTP requests where the client sent a RowId or other identifier that the server couldn't find. IllegalArgumentException would be more appropriate here.